gsitemap.php

```diff
@@ -1,907 +1,987 @@
-			if (isset($id_image['id_image']))
-			{
-				$image_link = $this->context->link->getImageLink($product->link_rewrite, $product->id.'-'.(int)$id_image['id_image'], 'large_default');
-				$image_link = (!in_array(rtrim(Context::getContext()->shop->virtual_uri, '/'), explode('/', $image_link))) ? str_replace(
-					array(
-						'https',
-						Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri
-					), array(
-						'http',
-						Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri.Context::getContext()->shop->virtual_uri
-					), $image_link
-				) : $image_link;
-			}
-			$file_headers = (Configuration::get('GSITEMAP_CHECK_IMAGE_FILE')) ? @get_headers($image_link) : true;
-			$image_product = array();
-			if (isset($image_link) && ($file_headers[0] != 'HTTP/1.1 404 Not Found' || $file_headers === true))
-				$image_product = array(
-					'title_img' => htmlspecialchars(strip_tags($product->name)),
-					'caption' => htmlspecialchars(strip_tags($product->description_short)),
-					'link' => $image_link
-				);
-			if (!$this->_addLinkToSitemap(
-				$link_sitemap, array(
-					'type' => 'product',
-					'page' => 'product',
-					'lastmod' => $product->date_upd,
-					'link' => $url,
-					'image' => $image_product
-				), $lang['iso_code'], $index, $i, $product_id['id_product']
-			))
-				return false;
-
-			unset($image_link);
-		}
-
-		return true;
-	}
-
-	/**
-	 * Hydrate $link_sitemap with categories link
-	 *
-	 * @param array  $link_sitemap contain all the links for the Google Sitemap file to be generated
-	 * @param string $lang         language of link to add
-	 * @param int    $index        index of the current Google Sitemap file
-	 * @param int    $i            count of elements added to sitemap main array
-	 * @param int    $id_category  category object identifier
-	 *
-	 * @return bool
-	 */
-	protected function _getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_category = 0)
-	{
-		$link = new Link();
-		if (method_exists('ShopUrl', 'resetMainDomainCache'))
-			ShopUrl::resetMainDomainCache();
-
-		$categories_id = Db::getInstance()->ExecuteS(
-			'SELECT c.id_category FROM `'._DB_PREFIX_.'category` c
+    const HOOK_ADD_URLS = 'gSitemapAppendUrls';
+
+    public $cron = false;
+    protected $sql_checks = array();
+
+    public function __construct()
+    {
+        $this->name = 'gsitemap';
+        $this->tab = 'seo';
+        $this->version = '4.0.0';
+        $this->author = 'Prestashop';
+        $this->need_instance = 0;
+        $this->module_key = '';
+        $this->bootstrap = true;
+
+        parent::__construct();
+
+        $this->displayName = $this->trans('Google sitemap', array(), 'Modules.Gsitemap.Admin');
+        $this->description = $this->trans('Generate your Google sitemap file', array(), 'Modules.Gsitemap.Admin');
+        
+        $this->confirmUninstall = $this->l('Are you sure you want to uninstall this module?');
```

Comment:
You can use `$this->trans` with the domain "messages":

```php
$this->trans('Are you sure you want to uninstall this module?', array(), 'messages');

Reply:
In that specific case, only the first parameter should be actually enough. :)

Reply:
It falls into _Modules.Legalcompliance.Admin_ domains and the _messages_ file. But this string will certainly be reused so I guess it could be localized in the _Admin.Notifications.Warning_ domains or the _Admin.Global_ ones. Unless it should only be localized in the Modules folder?

Reply:
Fixed, used: Admin.Notifications.Warning